### ISSUE TITLE
Not working under Kodi 18.4 & Debian 10

### ISSUE THREAD

**Issue by Ludo444** [chadparry/kodi-plugin.program.remote.control.browser#5]
Hi,
when trying to run addon nothing happens for me. Log states following issue:

```
2019-11-27 21:24:54.701 T:139738086364928   ERROR: usage: [-h] handle params
2019-11-27 21:24:54.701 T:139738086364928   ERROR: : error: unrecognized arguments: resume:false
2019-11-27 21:24:54.713 T:139738858358528   ERROR: GetDirectory - Error getting plugin://plugin.program.remote.control.browser/
2019-11-27 21:24:54.725 T:139739538524800   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.program.remote.control.browser/) failed
```

Also it complains about missing python dependencies (which are present) and Insufficient memory in addon settings.

**Comment by chadparry** [comment]
I haven't upgraded to Kodi 18 yet. I'll test it out and get back to you. (This will take a few weeks because of holidays.)

**Comment by kjetilk** [comment]
Just to say I'm seeing the same thing. Enjoy the holidays!

**Comment by chadparry** [comment]
I've created a temporary workaround, so the add-on will function. Upgrade to at least version 1.5.0. It's not updated in the main Kodi add-on repositories yet, so you'll need to download it from here: https://chad.parry.org/software/kodi/ .

**Comment by Ludo444** [comment]
Hi, first thanks, but sadly there's still some issue for me. I'm able to get to bookmarks menu now, but addon reports that `Browser reported error`

```
Failed to spawn browser, errno=1: /usr/bin/python /var/lib/kodi/.kodi/addons/plugin.program.remote.control.browser/browse.py --alsa-control Master --xdotool-path /usr/bin/xdotool --lirc-config /var/lib/kodi/.kodi/addons/plugin.program.remote.control.browser/resources/data/lircd/youtubetv.lirc -- /usr/bin/chromium 'file:///var/lib/kodi/.kodi/addons/plugin.program.remote.control.browser/resources/data/black.html?https%3A%2F%2Fwww.youtube.com%2Ftv'
```

**Comment by chadparry** [comment]
@Ludo444 , I'll use this ticket to track the remaining work for the `usage` error. Will you please file a separate ticket for tracking the browser failure? Also, here are some questions I have:

1. Does the error happen every time?

1. Do you have a browser at `/usr/bin/chromium`? In other words, what happens if you type: `ls -l /usr/bin/chromium`?

1. That is not one of the standard browser names, since `/usr/bin/chromium-browser` is more common. Did you change it yourself in the settings? Also, did you remove the browser arguments yourself?

**Comment by Ludo444** [comment]
Ok, nevermind, that was just because of lircd daemon was not running. Seems its needed, and my issue is/was related to debian bug - https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=932779

So browser starts now, that part is fine. :)

Is the intended behavior currently that the Kodi arrows doesn't work for controls? Because when using remote I hear Kodi menu sounds in background moving, but I don't control website.

**Comment by bbukh** [comment]
After fixing lirc issue mentioned by @Ludo444 above. I have the same issue (browser starts, but I cannot control with Kodi arrows). Below is a copy of the relevant part of the Kodi log file.

2020-03-25 15:32:57.504 T:140006924334848    INFO: Launching wrapper for browser: /usr/bin/firefox -no-remote -new-window 'file:///
home/kodi/.kodi/addons/plugin.program.remote.control.browser/resources/data/black.html?https%3A%2F%2Fwww.kanopy.com'
2020-03-25 15:32:58.342 T:140008737852672   DEBUG: ------ Window Init (Pointer.xml) ------
2020-03-25 15:32:58.507 T:140006910682880   DEBUG: CWebServer[8083]: request received for /jsonrpc
2020-03-25 15:32:58.783 T:140007771711232   DEBUG: Previous line repeats 1 times.
2020-03-25 15:32:58.784 T:140007771711232   DEBUG: BROWSER: Launching browser: /usr/bin/firefox -no-remote -new-window 'file:///hom
e/kodi/.kodi/addons/plugin.program.remote.control.browser/resources/data/black.html?https%3A%2F%2Fwww.kanopy.com'
2020-03-25 15:32:58.794 T:140007771711232   DEBUG: BROWSER: Searching for browser PID: /usr/bin/xdotool search --sync --onlyvisible
 --pid 1514
2020-03-25 15:32:59.565 T:140007771711232   DEBUG: BROWSER: X Error of failed request:  BadWindow (invalid Window parameter)
2020-03-25 15:32:59.565 T:140007771711232   DEBUG: BROWSER:   Major opcode of failed request:  15 (X_QueryTree)
2020-03-25 15:32:59.565 T:140007771711232   DEBUG: BROWSER:   Resource id in failed request:  0x3000002
2020-03-25 15:32:59.566 T:140007771711232   DEBUG: BROWSER:   Serial number of failed request:  806
2020-03-25 15:32:59.566 T:140007771711232   DEBUG: BROWSER:   Current serial number in output stream:  806
2020-03-25 15:32:59.566 T:140007771711232   DEBUG: BROWSER: Failed search for browser PID
2020-03-25 15:32:59.568 T:140007771711232   DEBUG: BROWSER: Exception in thread Thread-2:
2020-03-25 15:32:59.569 T:140007771711232   DEBUG: BROWSER: Traceback (most recent call last):
2020-03-25 15:32:59.569 T:140007771711232   DEBUG: BROWSER:   File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inne
r
2020-03-25 15:32:59.569 T:140007771711232   DEBUG: BROWSER:     self.run()
2020-03-25 15:32:59.569 T:140007771711232   DEBUG: BROWSER:   File "/usr/lib/python2.7/threading.py", line 754, in run
2020-03-25 15:32:59.570 T:140007771711232   DEBUG: BROWSER:     self.__target(*self.__args, **self.__kwargs)
2020-03-25 15:32:59.570 T:140007771711232   DEBUG: BROWSER:   File "/home/kodi/.kodi/addons/plugin.program.remote.control.browser/browse.py", line 296, in activateWindow
2020-03-25 15:32:59.572 T:140007771711232   DEBUG: BROWSER:     raise subprocess.CalledProcessError(proc.returncode, cmd, output)
2020-03-25 15:32:59.573 T:140007771711232   DEBUG: BROWSER: CalledProcessError: Command '['/usr/bin/xdotool', 'search', '--sync', '--onlyvisible', '--pid', '1514']' returned non-zero exit status 1
2020-03-25 15:32:59.573 T:140007771711232   DEBUG: BROWSER:
2020-03-25 15:33:00.099 T:140006910682880   DEBUG: CWebServer[8083]: request received for /jsonrpc
2020-03-25 15:33:02.495 T:140008737852672   DEBUG: Previous line repeats 3 times.
2020-03-25 15:33:02.496 T:140008737852672   DEBUG: ------ Window Deinit (Pointer.xml) ------
2020-03-25 15:33:03.225 T:140006910682880   DEBUG: CWebServer[8083]: request received for /jsonrpc
2020-03-25 15:33:19.262 T:140007369058048   DEBUG: Previous line repeats 19 times.
2020-03-25 15:33:19.262 T:140007369058048    INFO: JSONRPC Server: Disconnection detected
2020-03-25 15:33:20.105 T:140007369058048   DEBUG: JSONRPC Server: New connection detected
2020-03-25 15:33:20.105 T:140007369058048    INFO: JSONRPC Server: New connection added
2020-03-25 15:33:21.069 T:140007377450752   DEBUG: EventClient: button code 0 pressed

**Comment by chadparry** [comment]
The problem with browser controls is a separate issue. It appears to be related to a bug in `xdotool`: https://github.com/jordansissel/xdotool/issues/60. I can probably create a workaround.

**Comment by bdshields** [comment]
Hi,
So I've done a bit of exploration regarding the lirc signals going to kodi, it appears they have removed the "Lirc.Start" and "Lirc.Stop" functions. I hacked in a xbmcgui window to capture (and drop) the actions and that appears to be working.

It won't be easy, but I'm wondering if getting the actions from kodi might be a better approach than using pylirc.

**Comment by chadparry** [comment]
Thanks for tracking down the issue, @bdshields. That's inconvenient that they removed the LIRC builtins, and it didn't even make the release notes.  It was done here: https://github.com/xbmc/xbmc/pull/13761 . I'll look into solutions.

**Comment by bdshields** [comment]
Getting the events directly from Kodi has merit. You remove a dependency (pylirc), however replicating the "mode" feature of lirc isn't so easy.

**Comment by chadparry** [comment]
I just published v1.6.0 to my repository, (https://chad.parry.org/software/kodi/repository.chad.parry.org/repository.chad.parry.org-3.1.0.zip). It now opens an XBMC window to capture keys. Thanks for the help, @bdshields. I retained the LIRC dependency for now, because of its "delay" feature, and because that's the only way to support the addon's feature "Suspend Kodi During Browsing."
